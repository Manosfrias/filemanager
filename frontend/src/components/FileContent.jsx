import React, { useEffect } from "react";
import PropTypes from 'prop-types';

import { getFiles } from "../../utils/api/files";
import styles from "./FileContent.module.css";

function FileContent({ file = null }) {

  useEffect(() => {
    const loadFiles = async () => {
      try {
        const fetechData = await getFiles();
        console.log("fetechData", fetechData);

      } catch (error) {
        console.error(error);
      }
      
    };
    loadFiles()
  }, []);

  return (
    <div className={styles.FileContent}>
      <pre>
        {file?.content && atob(file.content)}
      </pre>
    </div>
  );
}

FileContent.propTypes = {
  file: PropTypes.object
};

export default FileContent;
