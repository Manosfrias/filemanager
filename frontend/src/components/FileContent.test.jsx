import React from 'react';
import { render } from '@testing-library/react';

import FileContent from './FileContent';

describe('FileContent Component', () => {
  it('should render defaults', () => {
    const { container } = render(
      <FileContent />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
