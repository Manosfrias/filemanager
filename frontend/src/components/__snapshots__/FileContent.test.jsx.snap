// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileContent Component should render defaults 1`] = `
<div
  class="FileContent"
>
  <pre />
</div>
`;
